### PR TITLE
feat: split bh into two

### PR DESCRIPTION
**Pull request by kelemeno**
# What ❔

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.

### CODE REVIEW

File: l1-contracts/contracts/bridgehub/IInteropCenter.sol [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    function proveL2LogInclusion(
+        uint256 _chainId,
+        uint256 _batchNumber,
+        uint256 _index,
+        L2Log memory _log,
+        bytes32[] calldata _proof
+    ) external view returns (bool);
+
+    function proveL1ToL2TransactionStatus(
```
Comment (StanislavBreadless) [inline l1-contracts/contracts/bridgehub/IInteropCenter.sol:34]: We have the same function on Bridgehub, what is our roadmap here? 

BTW, I am totally fine with having both in the current release but strongly announce that in the next upgrade they will be removed from the code. But in this case, please add easily visible comments in Bridgehub that tell that these methods should not be used

File: l1-contracts/contracts/bridgehub/InteropCenter.sol [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+    /// @notice the asset id of Eth. This is only used on L1.
+    bytes32 internal immutable ETH_TOKEN_ASSET_ID;
+
+    /// @notice all the ether and ERC20 tokens are held by NativeVaultToken managed by this shared Bridge.
+    address public assetRouter;
+
+    /// @notice The contract that stores the cross-chain message root for each chain and the aggregated root.
+    /// @dev Note that the message root does not contain messages from the chain it is deployed on. It may
+    /// be added later on if needed.
+    // IMessageRoot public messageRoot;
```
Comment (StanislavBreadless) [inline l1-contracts/contracts/bridgehub/InteropCenter.sol:50]: commented out code

File: l1-contracts/contracts/bridgehub/InteropCenter.sol [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+        address _assetRouter
+    )
+        external
+        // ICTMDeploymentTracker _l1CtmDeployer,
+        // IMessageRoot _messageRoot
+        onlyOwner
+    {
+        assetRouter = _assetRouter;
+        // l1CtmDeployer = _l1CtmDeployer;
+        // messageRoot = _messageRoot;
```
Comment (StanislavBreadless) [inline l1-contracts/contracts/bridgehub/InteropCenter.sol:113]: please remove commented out code + fix doc comments

File: l1-contracts/contracts/bridgehub/InteropCenter.sol [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+    /// @notice Used to forward a transaction on the gateway to the chains mailbox (from L1).
+    /// @param _chainId the chainId of the chain
+    /// @param _canonicalTxHash the canonical transaction hash
+    /// @param _expirationTimestamp the expiration timestamp for the transaction
+    function forwardTransactionOnGateway(
+        uint256 _chainId,
+        bytes32 _canonicalTxHash,
+        uint64 _expirationTimestamp
+    ) external override onlySettlementLayerRelayedSender {
+        require(L1_CHAIN_ID != block.chainid, "BH: not in sync layer mode");
```
Comment (StanislavBreadless) [inline l1-contracts/contracts/bridgehub/InteropCenter.sol:304]: please use custom errors

File: l1-contracts/contracts/bridgehub/InteropCenter.sol [hunk 5]
```diff
@@ -0,0 +1,400 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.24;
+
+// solhint-disable reason-string, gas-custom-errors
```
Comment (StanislavBreadless) [inline l1-contracts/contracts/bridgehub/InteropCenter.sol:5]: this is a new contract, so please remove this line to not forget about custom errors

File: l1-contracts/contracts/state-transition/chain-deps/facets/Mailbox.sol [hunk 6]
```diff
@@ -65,7 +66,7 @@ contract MailboxFacet is ZKChainBase, IMailbox {
     /// @inheritdoc IMailbox
     function bridgehubRequestL2Transaction(
         BridgehubL2TransactionRequest calldata _request
-    ) external onlyBridgehub returns (bytes32 canonicalTxHash) {
+    ) external onlyInteropCenter returns (bytes32 canonicalTxHash) {
```
Comment (StanislavBreadless) [inline l1-contracts/contracts/state-transition/chain-deps/facets/Mailbox.sol:69]: wont work during the upgrade (during the time when chains themselves are upgraded, but not the rest of the system)

File: system-contracts/contracts/L2GenesisUpgradeHelper.sol [hunk 7]
```diff
@@ -113,5 +113,17 @@ library L2GenesisUpgradeHelper {
                 additionalForceDeploymentsData.baseTokenAssetId
             )
         });
+
+        forceDeployments[1] = ForceDeployment({
```
Comment (StanislavBreadless) [inline system-contracts/contracts/L2GenesisUpgradeHelper.sol:117]:
```suggestion
        forceDeployments[4] = ForceDeployment({
```

File: l1-contracts/contracts/bridge/L1Nullifier.sol [hunk 8]
```diff
@@ -42,6 +43,9 @@ contract L1Nullifier is IL1Nullifier, ReentrancyGuard, Ownable2StepUpgradeable,
     /// @dev Bridgehub smart contract that is used to operate with L2 via asynchronous L2 <-> L1 communication.
     IBridgehub public immutable override BRIDGE_HUB;
 
+    /// @dev InteropCenter smart contract that is used to used to operate with L2 via asynchronous L2 <-> L1 communication.
```
[the remark this answers is not in the document]
Comment (kelemeno) [reply l1-contracts/contracts/bridge/L1Nullifier.sol:46]: I think you mentioned this before, and I don't think this happened now. 
But removing anyway.

**Pull request closed without merging**